enhancements/network/ovs-dynamic-cpu-pinning.md

```diff
@@ -0,0 +1,99 @@
+---
+title: ovs-dynamic-cpu-pinning
+authors:
+  - "@zeeke"
+reviewers:
+  - "@trozet"
+  - "@MarSik"
+approvers:
+  - TBD
+creation-date: 2023-03-31
+last-updated: 2023-03-31
+tracking-link:
+  - https://issues.redhat.com/browse/CNF-5977
+---
+
+# OVS Dynamic CPU Pinning
+
+## Summary
+
+Allow OVS daemons to run on non reserved CPU if the network workload requires
+cycles and non-reserved CPU are idle.
+
+## Motivation
+
+OVS runs at reserved cpus when an OCP node is configured cpuManagerPolicy to `static`. (TBD doclink)
+Typically the customers want to minimize the amount of reserved cpus and leave
+as much cpus as possible available for the business related workloads.
+
+We have seen in some customer cases that it is possible to disrupt both the node
+and the cluster when the networking load rises, because OVS cannot keep up with 
+the flow and starts removing routes from its tables. Which just makes the situation 
+worse as the amount of upcalls (software based processing when a connection is not 
+present in the OVS tables) increases and increases the cpu load even further.
+
+
+### User Stories
+
+> As an Openshift cluster administrator
+> I want to make the most of the node's CPUs
+> So that the cluster can endure a high network workload.
+
+### Goals
+
+Every cluster node that is configured with static CPU manager policy and
+a low number of reserved CPU does not experience any excessive latency 
+increase while dealing with high network loads.
+
+### Non-Goals
+
+TBD
+
+## Proposal A
+
+The main idea to resolve the issue described above is to allow OVS 
+to spread out to other cpus outside the reserved cpu pool. However 
+a care must be taken to not interfere with cpus that are assigned 
+to latency sensitive workloads.
+
+
+### Workflow Description
+
+The feature is triggered by Node Tuning Operator (NTO) when it detects the 
+performance conditions to turn it on. The following steps will be implemented
+to turn the feature on:
+- NTO's daemonset moves the ovs daemon to a systemd slice different than the default one.
+- NTO populates the `PerformanceProfile.Status.OvsSlice` field
+- Cluster Network Operator watches for PerformanceProfile status field and
+renders the `ovnkube-node` daemonset resource, adding `--ovs-cpu-pinning-slice <slice-name>` parameter.
+- ovnkube-node pods starts a goroutine that constantly monitors the CPU used by the 
```

Comment:
I think this could use a bit more verbosity. What is needed for the alignment and what does it mean to align?

Reply:
Thanks for pointing out. I'll detail this part